### ISSUE TITLE
Compiling multiple entry files with reaxt

### ISSUE THREAD

**Issue by emilingerslev** [awetzel/reaxt#8]
I got an elixir app where I need multiple react apps. That can be done with different `react.render :<target>`, but it will stil use the same js file.
Is it somehow possible to make separate js files for different parts of the app?

**Comment by awetzel** [comment]
I am very sorry for the late answer, yes of course it is possible, using webpack.config you can customize "entries" as you want. (see http://webpack.github.io/docs/multiple-entry-points.html)

For instance replace `entry: "./index"` with 

```javascript
 entry: {
        lib1: "./index1",
        lib2: "./index2"
}
```

Then the `require()` and codes of index1 and index2 will be compiled in each file separatedly. The only (big) constraint is that reaxt will automatically include in all your entries all the react components (files in the "component" dir). So if you want to partition well your code, it is important to not put everything into components.

you can also customize the output file name and directory : 

```
  output: {
    path: '/some/output/dir',
    filename: '[name].[hash].js', 
    publicPath: '/myprefix/'
  }
```

Then you only have to include `WebPack.file_of(:lib1)` and `WebPack.file_of(:lib2)` (maybe `"lib2`, not sure, please test), in your html file and serve the output dir with the correct prefix : 

```
plug Plug.Static, at: "/myprefix", from: "/some/output/dir"
```

**Comment by emilingerslev** [comment]
I know it is with time :) I'll look into testing this out. I'm not sure what you mean with all components will be included into all entries though?

Does this mean that even if I don't reference the files in the entries it will still be included?

**Comment by awetzel** [comment]
Yes exactly, to be clear, the entry `client_entry_addition.js` is added to every entries by reaxt. and this file  contains `require("./../../components/"+module)` in order to make reaxt work. https://github.com/awetzel/reaxt/blob/master/priv/commonjs_reaxt/client_entry_addition.js#L8

So that makes all components to be included in all entries, see : 
http://webpack.github.io/docs/configuration.html#automatically-created-contexts-defaults-module-xxxcontextxxx

**Comment by emilingerslev** [comment]
Is it then beyond the scope of Reaxt to make multiple entries each with their own separate group of components?
